kotlinKoans/src/main/java/com/epam/kotlinkoans/i_introduction/_3_Default_Arguments/n03DefaultArguments.kt

```diff
@@ -1,26 +1,14 @@
 package i_introduction._3_Default_Arguments
 
-import com.epam.kotlinkoans.i_introduction._3_Default_Arguments.JavaCode3
-import util.TODO
-import util.doc2
+var defaultNumber = 42
 
-fun todoTask3(): Nothing = TODO(
-    """
-        Task 3.
-        Several overloaded 'foo' functions in the class 'JavaCode3' can be replaced with one function in Kotlin.
-        Change the declaration of the function 'foo' in a way that makes the code using 'foo' compile.
-        You have to add 'foo' parameters and implement its body.
-        Uncomment the commented code and make it compile.
-    """,
-    documentation = doc2(),
-    references = { name: String -> JavaCode3().foo(name); foo(name) })
-
-fun foo(name: String): String = todoTask3()
+fun foo(name: String, number: Int = defaultNumber, toUpperCase: Boolean = false): String {
```

Comment:
in kotlin better leave default variable like hardcode in method declaration, then create new variable

Reply:
```suggestion
fun foo(name: String, number: Int = 42, toUpperCase: Boolean = false): String {
    return (if (toUpperCase) name.toUpperCase() else name) + number;
}
```